app/src/main/java/org/schabi/newpipe/player/helper/PlayerHelper.java

```diff
@@ -83,6 +98,58 @@ public static String resizeTypeOf(@NonNull final Context context,
         }
     }
 
+    @NonNull
+    public static String cacheKeyOf(@NonNull final StreamInfo info, @NonNull VideoStream video) {
+        return info.getUrl() + video.getResolution() + video.getFormat().getName();
```

Comment:
Just for the record. The cache key generation here might be too naive. Hopefully someone can find a better alternative, such as generating the `VideoStream` or `AudioStream` identifier in the extractor.

Reply:
So you think we should distinguish between audio and video data as well?

Reply:
Sure, I think each normal stream should have a unique id they can use regardless of their url, since it changes all the time.

Reply:
Yes that makes better sense.
This is why the UrlIdHandlers exist. Maybe these should be used at this point.